contracts/SpokePool.sol

```diff
@@ -599,38 +734,53 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         // Stores the equivalent amount to be sent by the relayer before fees have been taken out.
         if (maxTokensToSend == 0) return 0;
 
-        // todo: consider if this code block could be simplified. improve the branching comments and perhaps change the
-        // variable names.
+        // Derive the amount of the relay filled if the caller wants to send exactly `maxTokensToSend` tokens to
+        // the recipient. For example, if the user wants to send 10 tokens to the recipient, the full relay amount
+        // is 100, and the fee %'s total 5%, then this computation would return ~10.5, meaning that to fill 10.5/100
+        // of the full relay size, the caller would need to send 10 tokens to the user.
         fillAmountPreFees = _computeAmountPreFees(
             maxTokensToSend,
             (relayData.realizedLpFeePct + updatableRelayerFeePct)
         );
         // If user's specified max amount to send is greater than the amount of the relay remaining pre-fees,
         // we'll pull exactly enough tokens to complete the relay.
         uint256 amountToSend = maxTokensToSend;
-        if (relayData.amount - relayFills[relayHash] < fillAmountPreFees) {
-            fillAmountPreFees = relayData.amount - relayFills[relayHash];
+        uint256 amountRemainingInRelay = relayData.amount - relayFills[relayHash];
+        if (amountRemainingInRelay < fillAmountPreFees) {
+            fillAmountPreFees = amountRemainingInRelay;
+
+            // The user will fulfill the remainder of the relay, so we need to compute exactly how many tokens post-fees
+            // that they need to send to the recipient.
             amountToSend = _computeAmountPostFees(
                 fillAmountPreFees,
                 relayData.realizedLpFeePct + updatableRelayerFeePct
             );
         }
+
+        // `relayFills` keeps track of pre-fee fill amounts as a convenience to relayers who want to specify round
+        // numbers for the `maxTokensToSend` parameter or convenient numbers like 100 (i.e. relayers who will fully
+        // fill any relay up to 100 tokens, and partial fill with 100 tokens for larger relays).
         relayFills[relayHash] += fillAmountPreFees;
+
         // If relay token is weth then unwrap and send eth.
         if (relayData.destinationToken == address(weth)) {
-            // Note: WETH is already in the contract in the slow relay case.
+            // Note: `useContractFunds` is True if we want to send funds to the recipient directly out of this contract,
+            // otherwise we expect the caller to send funds to the recipient. If `useContractFunds` is True and the
```

Comment:
```suggestion
            // Note: useContractFunds is True if we want to send funds to the recipient directly out of this contract,
            // otherwise we expect the caller to send funds to the recipient. If useContractFunds is True and the
```